sklearn/tree/_tree.pyx

```diff
@@ -391,29 +548,49 @@ cdef class Entropy(ClassificationCriterion):
 
         cdef double entropy = 0.0
         cdef double total = 0.0
-        cdef double tmp
+        cdef double pmk
         cdef SIZE_t k
         cdef SIZE_t c
 
+        # Go through each response and calculate the entropy
         for k in range(n_outputs):
+            # Initialize the entropy to 0.0 for this response
```

Comment:
I am not sure what you think about this, but I would rather remove verbose comments like this. The line below speaks for itself.